modules/standard/Random.chpl

```diff
@@ -300,6 +300,97 @@ module Random {
     compilerError("'shuffle' only supports 1D rectangular arrays");
   }
 
+  /*
+    Produce a random permutation of an array's elements
+
+    :arg arr: A 1D rectangular array
+    :arg seed: The seed to use when creating the ``randomStream``
+
+    :return: A new array containing each of the values from ``arr`` in a
+              pseudo-random order.
+  */
+  proc permute(const ref arr: [?d] ?t, seed: int): [] t
+    where is1DRectangularDomain(d)
+  {
+    var rs = new randomStream(d.idxType, seed);
+    return rs.permute(arr);
+  }
+
+  /*
+    Produce a random permutation of an array's elements
+
+    :arg arr: A 1D rectangular array
+
+    :return: A new array containing each of the values from ``arr`` in a
+              pseudo-random order.
+  */
+  @unstable("the overload of permute that generates its own seed is unstable")
+  proc permute(const ref arr: [?d] ?t): [] t
+    where is1DRectangularDomain(d)
+  {
+    var rs = new randomStream(d.idxType);
+    return rs.permute(arr);
+  }
+
+  /*
+    Produce a random permutation of the indices in a domain.
+
+    :arg d: A 1D rectangular domain
+    :arg seed: The seed to use when creating the ``randomStream``
+
+    :return: An array containing each of the indices from ``d`` in a
+              pseudo-random order.
+  */
+  proc permute(d: domain(?), seed: int): [] d.idxType
+    where is1DRectangularDomain(d)
+  {
+    var rs = new randomStream(d.idxType, seed);
+    return rs.permute(d);
+  }
+
+  /*
+    Produce a random permutation of the indices in a domain.
+
+    :arg d: A 1D rectangular domain
+
+    :return: An array containing each of the indices from ``d`` in a
+              pseudo-random order.
+  */
+  @unstable("the overload of permute that generates its own seed is unstable")
```

Comment:
nit, put permute in single quotes as `'permute'` to match the other unstable warnings